movement/watch_faces/complication/activity_face.c

```diff
@@ -0,0 +1,714 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/* ** TODO
+ * ===================
+ * -- Reset length limit to 5 min :)
+ * -- Additional power-saving optimizations
+ * OK Save logged activity; restore real length limit
+ * OK FULL behavior
+ * OK Clear blinking confirmation
+ * OK Actually clear
+ * OK Actually chirp
+ * OK Display countdown while chirping
+ * OK Quit chirp by mode press
+ * OK Update demo face with correct prefix
+ * OK Run on 1-second tick
+ * OK Fast blink for CLEAR confirm
+ * OK Low energy mode > movement_le_inactivity_deadlines
+ * -- Clear confirm: faster animation
+ * -- Write in-comment documentation
+ * OK Stop at 8 hours
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include "activity_face.h"
+#include "chirpy_tx.h"
+#include "watch.h"
+#include "watch_utility.h"
+
+// ===========================================================================
+// This part is configurable: you can edit values here to customize you activity face
+// In particular, with num_enabled_activities and enabled_activities you can choose a subset of the
+//   activities that you want to see in your watch.
+// You can also add new items to activity_names, but don't redefine or remove existing ones.
+
+// If a logged activity is shorter than this, then it won't be added to log when it ends.
+// This way scarce log slots are not taken up by aborted events that weren't real activities.
+static const uint16_t activity_min_length_sec = 10;  // 300
+
+// Supported activities. ID of activity is index in this buffer
+// W e should never change order or redefine items, only add new items when needed.
+static const char activity_names[][14] = {
+    " bIKE ",
+    "uuaLK ",
+    "  rUn ",
+    "DAnCE ",
+    " yOgA ",
+    "CrOSS ",
+    "Suuinn",
+    "ELLIP ",
+    "  gYnn",
+    "  rOuu",
+    "SOCCEr",
+    " FOOTb",
+    " bALL ",
+    "  SKI ",
+};
+
+// Number of currently enabled activities (size of enabled_activities).
+static const uint8_t num_enabled_activities = 14;
```

Comment:
you can make this dynamic with something like:
`static const uint8_t num_enabled_activities = sizeof(activity_names) / sizeof(*activity_names);`

Reply:
oh actually I see that it's actually supposed to be *enabled* activities. in that case you can just do a `sizeof(enabled_activities)` (and move this to after the enabled_activities array too obviously)

Reply:
Thanks, great point, I didn't know I could this in C. Fixed; pushing commit in a bit.